### PR TITLE
azure: stop on first error, qa: change deprecated --std=posix option to --library=posix

### PR DESCRIPTION
**Pull request by andreasgerstmayr**

### CODE REVIEW

File: src/include/builddefs.in [hunk 1]
```diff
@@ -175,7 +175,7 @@ ECHO	= @echo@
 LN_S	= @LN_S@
 GREP    = @grep@
 GIT	= @GIT@
-CLINT	= @CPPCHECK@ --std=posix --force --quiet --error-exitcode=1 -I$(TOPDIR)/src/include/pcp -I/usr/include/bits
+CLINT	= @CPPCHECK@ --library=posix --force --quiet --error-exitcode=1 -I$(TOPDIR)/src/include/pcp -I/usr/include/bits
```
Comment (lberk) [inline src/include/builddefs.in]: Have we checked how far back this change exists?  el6 era hosts? older debian distros? BSDs?
@kmcdonell any insight into if we'll need a check for this flag?

Comment (andreasgerstmayr) [reply src/include/builddefs.in]: Oops, I thought about it, but then forgot. What's a quick way to check this? Looking up the cppcheck source code, finding the version which introduced the `--library=posix` flag, and then check all 100 package managers and distro versions sounds a bit tedious :|

(The most obvious way would be to integrate all supported platforms into azure pipelines ;))

**Comment by kmcdonell** [comment]
On 11/7/19 12:41 am, Lukas Berk wrote:
> *@lberk* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In src/include/builddefs.in 
> <https://github.com/performancecopilot/pcp/pull/694#discussion_r302102577>:
> 
>> @@ -175,7 +175,7 @@ ECHO	= @echo@
>   LN_S	= @LN_S@
>   GREP    = @grep@
>   GIT	= @GIT@
> -CLINT	= @CPPCHECK@ --std=posix --force --quiet --error-exitcode=1 -I$(TOPDIR)/src/include/pcp -I/usr/include/bits
> +CLINT	= @CPPCHECK@ --library=posix --force --quiet --error-exitcode=1 -I$(TOPDIR)/src/include/pcp -I/usr/include/bits
> 
> Have we checked how far back this change exists? el6 era hosts? older 
> debian distros? BSDs?
> @kmcdonell <https://github.com/kmcdonell> any insight into if we'll need 
> a check for this flag?

It looks like this is a recent change in the fedora ecosystem ... Ubuntu 
19.04 (latest) uses --std.

I recommend dealing with this the same way we handle similar cases 
elsewhere in the code base ... in configure run cppcheck --help then 
grep the output to set something log CPPCHECK_POSIX_OPT to either 
--std=posix or --library=posix

**Comment by natoscott** [comment]
@andreasgerstmayr @kmcdonell I've added the CPPCHECK_POSIX_OPT logic and merged now.

**Pull request merged**